linalg/sundials.cpp

```diff
@@ -1573,11 +1589,19 @@ void ARKStepSolver::Init(TimeDependentOperator &f_)
       MFEM_VERIFY(sundials_mem, "error in ARKStepCreate()");
 
       // Attach the ARKStepSolver as user-defined data
+#if MFEM_SUNDIALS_VERSION < 70100
       flag = ARKStepSetUserData(sundials_mem, this);
+#else
+      flag = ARKodeSetUserData(sundials_mem, this);
+#endif
       MFEM_VERIFY(flag == ARK_SUCCESS, "error in ARKStepSetUserData()");
```

Comment:
Minor point, but this verify statement will technically be incorrect with SUNDIALS 7 (in that it'll show `ARKStepSetUserData` instead of `ARKodeSetUserData`).  There are various other verify statements with the same issue.

Instead of duplicating the verify statements in each of these instances, I suppose there are two options that have been in the past:
1. Use a macro to handle the function prefix mangling, i.e., something like
``` c++
#if MFEM_SUNDIALS_VERSION < 70100 
#define ARKodeORstep(FUNC_NAME) ARKode##FUNC_NAME
#else
#define ARKodeORstep(FUNC_NAME) ARKStep##FUNC_NAME
#endif

flag = ARKodeORstep(SetUserData)(sundials_mem, this);
MFEM_VERIFY(flag == ARK_SUCCESS, "error in ARKodeORstep(SetUserData)"
```
2. Introduce wrapper functions for the ones that don't exist in prior version (see, e.g., line 98), i.e., something like
```c++
#if MFEM_SUNDIALS_VERSION < 70100 
int ARKodeSetUserData(void* sundials_mem, void* something)
{
  return ARKStepSetUserData(sundials_mem, something);
}
#endif

flag = ARKodeSetUserData(sundials_mem, this);
MFEM_VERIFY(flag == ARK_SUCCESS, "error in ARKodeSetUserData()");
```

Personally, I think the second option is cleaner to read in the class implementations but does require more wrapper definitions at the top.